clustering/common/src/main/java/org/jboss/as/clustering/controller/AddStepHandler.java

```diff
@@ -90,11 +63,31 @@ protected void rollbackRuntime(OperationContext context, ModelNode operation, Re
     }
 
     @Override
+    protected void populateModel(final ModelNode operation, final ModelNode model) throws OperationFailedException {
+        for (AttributeDefinition attribute : this.descriptor.getAttributes()) {
+            attribute.validateAndSet(operation, model);
+        }
+    }
+
+    @Override
     public void register(ManagementResourceRegistration registration) {
-        SimpleOperationDefinitionBuilder builder = new SimpleOperationDefinitionBuilder(ModelDescriptionConstants.ADD, this.resolver).withFlag(OperationEntry.Flag.RESTART_NONE);
-        for (Attribute attribute : this.attributes) {
-            builder.addParameter(attribute.getDefinition());
+        SimpleOperationDefinitionBuilder builder = new SimpleOperationDefinitionBuilder(ModelDescriptionConstants.ADD, this.descriptor.getDescriptionResolver()).withFlag(OperationEntry.Flag.RESTART_NONE);
+        for (AttributeDefinition attribute : this.descriptor.getAttributes()) {
+            builder.addParameter(attribute);
+        }
+        for (AttributeDefinition parameter : this.descriptor.getExtraParameters()) {
+            builder.addParameter(parameter);
         }
         registration.registerOperationHandler(builder.build(), this);
     }
+
+    @Override
+    protected void recordCapabilitiesAndRequirements(OperationContext context, ModelNode operation, Resource resource) throws OperationFailedException {
+        PathAddress address = context.getCurrentAddress();
+        // The super implementation assumes that the capability name is a simple extension of the base name - we do not.
+        for (Capability capability : this.descriptor.getCapabilities()) {
+            context.registerCapability(capability.getRuntimeCapability(address), null);
+        }
```

Comment:
Oh crap.  You're right.  The super implementation iterates over its attribute collection - and we aren't populating that.  Will fix.